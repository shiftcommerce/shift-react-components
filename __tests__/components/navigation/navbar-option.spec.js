import React from 'react'

// Libraries
import componentMapping from '../../../src/lib/component-mapping'

// Components
import NavBarOption from '../../../src/components/navigation/navbar-option'

test('renders the links', () => {
  // Arrange
  const emptyFunction = () => { }
  const Link = componentMapping('Link')
  // Act
  const wrapper = shallow(
    <NavBarOption title='Mens' href='/slug?slug=/categories/mens' as='/categories/mens' onClick={emptyFunction} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toContainReact(<Link href='/slug?slug=/categories/mens' as='/categories/mens' className='c-nav__option' onClick={emptyFunction} ><div className='c-nav__option-label'>Mens</div><div className='c-nav__option-text'>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</div></Link>)
})
