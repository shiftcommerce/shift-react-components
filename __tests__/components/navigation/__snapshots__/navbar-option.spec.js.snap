// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the links 1`] = `
<Link
  as="/categories/mens"
  className="c-nav__option"
  href="/slug?slug=/categories/mens"
  onClick={[Function]}
>
  <div
    className="c-nav__option-label"
  >
    Mens
  </div>
  <div
    className="c-nav__option-text"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </div>
</Link>
`;
